### DISCUSSION TITLE
EBA DEMO | Observation of difference b/w Signify & NordVlei extensions

### DISCUSSION THREAD

**Discussion by HunnySajid** [WebOfTrust/polaris-web#30] [Q&A] [answered]
**Scenario:**
We are required to test nord vlei & signify browser extension compatibility with [reg-pilot-webapp](https://github.com/GLEIF-IT/reg-pilot-webapp). 
During this testing I observed a couple of issues:

**1- Nord extension throws following errors**

<img width="976" alt="Screenshot 2024-10-14 at 5 18 32 PM" src="https://github.com/user-attachments/assets/bb63ee5c-78d9-4107-88da-5302691bec57">

As per my understanding from debugging, it is listening to multiple events that are broadcasted from other extensions.
Following image shows a payload that is from metamask:
<img width="976" alt="Screenshot 2024-10-14 at 5 20 31 PM" src="https://github.com/user-attachments/assets/c1a126f0-e4d3-4ced-b69b-d5ad6968f60e">

When I tried it in an isolated browser (without metamask) it worked fine.

**2- RootsID domain whitelisting**
Currently, nord extension content_script only allows certain domains. 

It would be awesome if rootsID can get their domain whitelisted so that Nord extension can be demoed on deployed web apps.
Or rootsID can create certain domains following this schema that is acceptable.

Following are the rootsID webapp dev & test environment:
https://reg-pilot-webapp-dev.rootsid.cloud/
https://reg-pilot-webapp-test.rootsid.cloud/

**3- Session Visualization**
In signify extension, when a credential is selected then a session is created that lives for specific time.
This session info is also communicated to the web app and a UI also rendered by content script to let webapp know that
it is connected with signify extension and can receive signed header upon request.
<img width="99" alt="Screenshot 2024-10-14 at 6 05 28 PM" src="https://github.com/user-attachments/assets/e69b008b-e521-4b3e-82e5-101005dcede3">

In polaris web there is a method named `getSessionInfo` that is used to fetch session info.
Example usage:

```
const populateExtensionStatus = async () => {
    const extensionId = await signifyClient.isExtensionInstalled();
    setExtensionInstalled(Boolean(extensionId));
    if (extensionId) {
      console.log("extensionId", extensionId);
      try {
        const sessionObj = await signifyClient.getSessionInfo();
        await handleSignifyData(sessionObj);
        console.log("sessionObj", sessionObj);
        setSessionInfoLoaded(true);
      } catch (error) {
        console.log("error", error);
        setSessionInfoLoaded(true);
      }
    } else {
      setSessionInfoLoaded(true);
    }
  };
```

As it is not implemented in nord extension so this call does not receive a response.
We have used a custom timeout for now to deal with it:

```
const getSessionInfo = async () => {
    return new Promise(async (resolve, reject) => {
      const timer = setTimeout(() => {
        reject(null);
      }, 3000);
  
      const sessionObj = await signifyClient.getSessionInfo();
      clearTimeout(timer);
      resolve(sessionObj);
    })
  };
```

Nord could provide insight on whether they plan to implement this call during their extension.
I’m looking for input on these ideas and any other potential use cases or challenges this might raise.

@lenkan @2byrds

**Comment by 2byrds** [comment]
@lenkan if you interested in scheduling a meeting to discuss i'll be happy to coordinate that or we are flexible to discuss it asynchronously here via this github discussion

**Comment by lenkan** [comment, marked as answer]
Thanks for the mention @HunnySajid.

> 1- Nord extension throws following errors

This has now been resolved in v0.0.8.

> 2- RootsID domain whitelisting

I will bring this up internally. Ideally, I think the users should be in control of the whitelisting themselves. They could enable it for each domain manually.

> 3- Session Visualization

Great feedback! I have just published v0.0.8. This version is sending a `/signify/reply` back with an error if we are receiving a message of type `/signify/*` that we have not yet implemented. So it should reject immediately now.

We are planning to add support for the getSessionInfo feature. But because none of our users are asking for it, it is not in our top priority at the moment.
